web-wallet/src/lib/dusk/string/validateAddress.js

```diff
@@ -0,0 +1,20 @@
+/**
+ * Validates a Dusk address, with feedback on failure or success.
+ * @param {String} address The public spent key to validate.
+ * @returns {{isValid: boolean, reason: string}} An object with two keys:
+ *  - `isValid` {Boolean} - true if the address is valid, false if invalid.
+ *  - `reason` {String} - describes why the address is invalid or confirms if it is valid.
+ */
+export default function validateAddress (address) {
+	const regex = /[^\w]/g;
```

Comment:
This way you are including invalid characters though.

Write some expectations failing with the most common ones like `O`, `0` `l` and `I`.

(also: no need to define the regex here as it's not used elsewhere: just pass it to the search method, and no need to use the `g` flag)

Reply:
> So we are sure that no address can contain O, 0 l or I?

That I don't know: it's what @HDauven said.

> What are others invalid characters you can think of that you would like me to include?
> 
> Generally, when you mentioned this approach, I was worried that blacklisting all invalid characters might be trickier than whitelisting all the valid ones.

I'm sorry if I gave that impression, but I meant no such thing: I would, as you did, use a negated character class (`[^xyx]`), as it's easier.
I was just suggesting to use ranges, e.g. `[^1-9A-HJ-NP-Za-km-z]` (don't take this for granted as I wrote it here without checking).

Reply:
> From what I can see on the web, Bitcoin addresses use Base58Check which excludes "0" (zero), "O" (uppercase o), "l" (lowercase L), and "I" (uppercase i). Just wanted to confirm we do the same?

That's what was in his original regex

Reply:
We indeed adhere to the Bitcoin Base58Check AFAIK

Reply:
> (also: no need to define the regex here as it's not used elsewhere: just pass it to the search method, and no need to use the g flag)

I prefer to have the regex defined as a const within the function, as it improves readability IMO.

I have removed the `g` flag tho.